packages/firebase/index.d.ts

```diff
@@ -1580,14 +1580,18 @@ declare namespace firebase.appCheck {
   export interface AppCheck {
     /**
      * Activate AppCheck
-     * @param provider reCAPTCHA or custom token provider.
+     * @param provider reCAPTCHA provider, custom token provider, or reCAPTCHA site key.
      * @param isTokenAutoRefreshEnabled If true, the SDK automatically
      * refreshes App Check tokens as needed. If undefined, defaults to the
      * value of `app.automaticDataCollectionEnabled`, which defaults to
      * false and can be set in the app config.
      */
     activate(
-      provider: AppCheckProvider,
+      provider:
+        | ReCaptchaV3Provider
+        | CustomProvider
+        | AppCheckProvider
```

Comment:
There is no need to list the base type if we are enumerating all specific providers. That means we can just remove AppCheckProvider from the public API.

Reply:
We were trying to make activate() completely backwards compatible, and the old API was that you can supply a siteKey (string) or for a custom provider you can supply an `AppCheckProvider` that only needs to be an object with a `getToken()` method, so I think that type has to stay. The new CustomProvider doesn't have the same API.